### ISSUE TITLE
[Bug]: Querycoord can not recover after querycoord pod kill chaos has been deleted

### ISSUE THREAD

**Issue by zhuwenxing** [milvus-io/milvus#18706]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:2.1.0-20220817-1e1b63d8
- Deployment mode(standalone or cluster): cluster
- SDK version(e.g. pymilvus v2.0.0rc2):pymilvus==2.2.0.dev8
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

Querycoord can not recover after querycoord pod kill chaos has been deleted

![image](https://user-images.githubusercontent.com/12268675/185307232-4c6586cc-0aa0-4677-bb35-23b7e5f1c7e1.png)

### Expected Behavior

all pods works well after chaos deleted

### Steps To Reproduce

```markdown
1. Deploy Milvus
2. Run e2e test before chaos
3. Run hello_milvus before chaos
4. Run chaos test and send request
```

### Milvus Log

failed job: chaos-test/496/pipeline
log:
[artifacts-querycoord-pod-kill-496-server-logs.tar.gz](https://github.com/milvus-io/milvus/files/9368107/artifacts-querycoord-pod-kill-496-server-logs.tar.gz)

[artifacts-querycoord-pod-kill-496-pytest-logs.tar.gz](https://github.com/milvus-io/milvus/files/9368109/artifacts-querycoord-pod-kill-496-pytest-logs.tar.gz)

**Comment by zhuwenxing** [comment]
```
[2022/08/18 04:22:32.043 +00:00] [ERROR] [querycoord/task_scheduler.go:265] ["reloadFromKV: taskStateInfo and triggerTaskInfo are inconsistent"] [stack="github.com/milvus-io/milvus/internal/querycoord.(*TaskScheduler).reloadFromKV\n\t/go/src/github.com/milvus-io/milvus/internal/querycoord/task_scheduler.go:265\ngithub.com/milvus-io/milvus/internal/querycoord.newTaskScheduler\n\t/go/src/github.com/milvus-io/milvus/internal/querycoord/task_scheduler.go:200\ngithub.com/milvus-io/milvus/internal/querycoord.(*QueryCoord).Init.func1\n\t/go/src/github.com/milvus-io/milvus/internal/querycoord/query_coord.go:212\nsync.(*Once).doSlow\n\t/usr/local/go/src/sync/once.go:68\nsync.(*Once).Do\n\t/usr/local/go/src/sync/once.go:59\ngithub.com/milvus-io/milvus/internal/querycoord.(*QueryCoord).Init\n\t/go/src/github.com/milvus-io/milvus/internal/querycoord/query_coord.go:135\ngithub.com/milvus-io/milvus/internal/distributed/querycoord.(*Server).init\n\t/go/src/github.com/milvus-io/milvus/internal/distributed/querycoord/service.go:227\ngithub.com/milvus-io/milvus/internal/distributed/querycoord.(*Server).Run\n\t/go/src/github.com/milvus-io/milvus/internal/distributed/querycoord/service.go:96\ngithub.com/milvus-io/milvus/cmd/components.(*QueryCoord).Run\n\t/go/src/github.com/milvus-io/milvus/cmd/components/query_coord.go:50\ngithub.com/milvus-io/milvus/cmd/roles.(*MilvusRoles).runQueryCoord.func1\n\t/go/src/github.com/milvus-io/milvus/cmd/roles/roles.go:186"]
[2022/08/18 04:24:03.057 +00:00] [WARN] [roles/healthz_handler.go:100] ["component is unhealthy"] [state=Initializing]
[2022/08/18 04:24:10.592 +00:00] [WARN] [roles/healthz_handler.go:100] ["component is unhealthy"] [state=Initializing]
[2022/08/18 04:24:20.593 +00:00] [WARN] [roles/healthz_handler.go:100] ["component is unhealthy"] [state=Initializing]
[2022/08/18 04:24:30.592 +00:00] [WARN] [roles/healthz_handler.go:100] ["component is unhealthy"] [state=Initializing]
[2022/08/18 04:24:30.593 +00:00] [WARN] [roles/healthz_handler.go:100] ["component is unhealthy"] [state=Initializing]
[2022/08/18 04:24:40.593 +00:00] [WARN] [roles/healthz_handler.go:100] ["component is unhealthy"] [state=Initializing]
[2022/08/18 04:24:50.593 +00:00] [WARN] [roles/healthz_handler.go:100] ["component is unhealthy"] [state=Initializing]
[2022/08/18 04:25:00.592 +00:00] [WARN] [roles/healthz_handler.go:100] ["component is unhealthy"] [state=Initializing]
[2022/08/18 04:25:00.593 +00:00] [WARN] [roles/healthz_handler.go:100] ["component is unhealthy"] [state=Initializing]
[2022/08/18 04:25:10.593 +00:00] [WARN] [roles/healthz_handler.go:100] ["component is unhealthy"] [state=Initializing]
[2022/08/18 04:25:20.057 +00:00] [WARN] [roles/healthz_handler.go:100] ["component is unhealthy"] [state=Initializing]
[2022/08/18 04:25:20.593 +00:00] [WARN] [roles/healthz_handler.go:100] ["component is unhealthy"] [state=Initializing]
[2022/08/18 04:25:30.593 +00:00] [WARN] [roles/healthz_handler.go:100] ["component is unhealthy"] [state=Initializing]
[2022/08/18 04:25:30.593 +00:00] [WARN] [roles/healthz_handler.go:100] ["component is unhealthy"] [state=Initializing]
[2022/08/18 04:25:40.593 +00:00] [WARN] [roles/healthz_handler.go:100] ["component is unhealthy"] [state=Initializing]
[2022/08/18 04:25:50.593 +00:00] [WARN] [roles/healthz_handler.go:100] ["component is unhealthy"] [state=Initializing]
[2022/08/18 04:26:00.593 +00:00] [WARN] [roles/healthz_handler.go:100] ["component is unhealthy"] [state=Initializing]
[2022/08/18 04:26:00.593 +00:00] [WARN] [roles/healthz_handler.go:100] ["component is unhealthy"] [state=Initializing]
[2022/08/18 04:26:10.592 +00:00] [WARN] [roles/healthz_handler.go:100] ["component is unhealthy"] [state=Initializing]
[2022/08/18 04:26:20.592 +00:00] [WARN] [roles/healthz_handler.go:100] ["component is unhealthy"] [state=Initializing]
[2022/08/18 04:26:23.056 +00:00] [WARN] [roles/healthz_handler.go:100] ["component is unhealthy"] [state=Initializing]
[2022/08/18 04:26:30.593 +00:00] [WARN] [roles/healthz_handler.go:100] ["component is unhealthy"] [state=Initializing]
[2022/08/18 04:26:30.593 +00:00] [WARN] [roles/healthz_handler.go:100] ["component is unhealthy"] [state=Initializing]
[2022/08/18 04:26:30.594 +00:00] [WARN] [roles/healthz_handler.go:100] ["component is unhealthy"] [state=Initializing]
[2022/08/18 04:26:30.607 +00:00] [ERROR] [roles/roles.go:524] ["Get signal to exit\n"] [signal=terminated] [stack="github.com/milvus-io/milvus/cmd/roles.(*MilvusRoles).Run\n\t/go/src/github.com/milvus-io/milvus/cmd/roles/roles.go:524\ngithub.com/milvus-io/milvus/cmd/milvus.(*run).execute\n\t/go/src/github.com/milvus-io/milvus/cmd/milvus/run.go:111\ngithub.com/milvus-io/milvus/cmd/milvus.RunMilvus\n\t/go/src/github.com/milvus-io/milvus/cmd/milvus/milvus.go:60\nmain.main\n\t/go/src/github.com/milvus-io/milvus/cmd/main.go:26\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:225"]
[2022/08/18 04:26:30.607 +00:00] [DEBUG] [querycoord/service.go:283] ["QueryCoord stop"] [Address=10.102.8.65:19531]
[2022/08/18 04:26:30.608 +00:00] [INFO] [querycoord/query_coord.go:276] ["close channel unsubscribe loop..."]
[2022/08/18 04:26:30.608 +00:00] [INFO] [querycoord/query_coord.go:281] ["cancel the loop of QueryCoord..."]
[2022/08/18 04:26:30.608 +00:00] [INFO] [querycoord/query_coord.go:285] ["Query Coord stopped successfully..."]
[2022/08/18 04:26:30.608 +00:00] [DEBUG] [querycoord/service.go:295] ["Graceful stop grpc server..."]

```

**Comment by zhuwenxing** [comment]
It reproduced stably

**Comment by zhuwenxing** [comment]
Not sure whether it is related to https://github.com/milvus-io/milvus/issues/18608

**Comment by jiaoew1991** [comment]
> Not sure whether it is related to #18608

It's different

**Comment by jiaoew1991** [comment]
https://github.com/milvus-io/milvus/blob/90de312d4be68dd68e3cdf868cf4a19c9647eacf/internal/querycoord/task_scheduler.go#L106-L119

at Line116, if the taskChan is full the goroutine will be blocked

in this case there are lots of task generated before `QueryCoord` restart, after `QueryCoord` restart it read the tasks from Etcd which are more than 1024, the initial process blocked.

**Comment by jiaoew1991** [comment]
we should make addTask in an independent goroutine

**Comment by jiaoew1991** [comment]
@zhuwenxing  you can try again with pr #18801

**Comment by zhuwenxing** [comment]
Not reproduced yet, remove the critical label, but keep watching

**Comment by zhuwenxing** [comment]
It reproduced again in version `2.1.0-20220830-a926a7d2`. it seems that the error is same as before.
failed job: https://qa-jenkins.milvus.io/blue/organizations/jenkins/chaos-test-for-release/detail/chaos-test-for-release/24/pipeline
log: 
[artifacts-querycoord-pod-kill-24-server-logs (1).tar.gz](https://github.com/milvus-io/milvus/files/9457813/artifacts-querycoord-pod-kill-24-server-logs.1.tar.gz)
[artifacts-querycoord-pod-kill-24-pytest-logs.tar.gz](https://github.com/milvus-io/milvus/files/9457814/artifacts-querycoord-pod-kill-24-pytest-logs.tar.gz)


```
[2022/08/30 14:30:26.372 +00:00] [ERROR] [querycoord/task_scheduler.go:265] ["reloadFromKV: taskStateInfo and triggerTaskInfo are inconsistent"] [stack="github.com/milvus-io/milvus/internal/querycoord.(*TaskScheduler).reloadFromKV\n\t/go/src/github.com/milvus-io/milvus/internal/querycoord/task_scheduler.go:265\ngithub.com/milvus-io/milvus/internal/querycoord.newTaskScheduler\n\t/go/src/github.com/milvus-io/milvus/internal/querycoord/task_scheduler.go:200\ngithub.com/milvus-io/milvus/internal/querycoord.(*QueryCoord).Init.func1\n\t/go/src/github.com/milvus-io/milvus/internal/querycoord/query_coord.go:212\nsync.(*Once).doSlow\n\t/usr/local/go/src/sync/once.go:68\nsync.(*Once).Do\n\t/usr/local/go/src/sync/once.go:59\ngithub.com/milvus-io/milvus/internal/querycoord.(*QueryCoord).Init\n\t/go/src/github.com/milvus-io/milvus/internal/querycoord/query_coord.go:135\ngithub.com/milvus-io/milvus/internal/distributed/querycoord.(*Server).init\n\t/go/src/github.com/milvus-io/milvus/internal/distributed/querycoord/service.go:227\ngithub.com/milvus-io/milvus/internal/distributed/querycoord.(*Server).Run\n\t/go/src/github.com/milvus-io/milvus/internal/distributed/querycoord/service.go:96\ngithub.com/milvus-io/milvus/cmd/components.(*QueryCoord).Run\n\t/go/src/github.com/milvus-io/milvus/cmd/components/query_coord.go:50\ngithub.com/milvus-io/milvus/cmd/roles.(*MilvusRoles).runQueryCoord.func1\n\t/go/src/github.com/milvus-io/milvus/cmd/roles/roles.go:186"]
[2022/08/30 14:31:55.483 +00:00] [WARN] [roles/healthz_handler.go:100] ["component is unhealthy"] [state=Initializing]
[2022/08/30 14:31:55.483 +00:00] [WARN] [roles/healthz_handler.go:100] ["component is unhealthy"] [state=Initializing]
[2022/08/30 14:32:05.483 +00:00] [WARN] [roles/healthz_handler.go:100] ["component is unhealthy"] [state=Initializing]
[2022/08/30 14:32:15.483 +00:00] [WARN] [roles/healthz_handler.go:100] ["component is unhealthy"] [state=Initializing]
[2022/08/30 14:32:25.483 +00:00] [WARN] [roles/healthz_handler.go:100] ["component is unhealthy"] [state=Initializing]
[2022/08/30 14:32:25.483 +00:00] [WARN] [roles/healthz_handler.go:100] ["component is unhealthy"] [state=Initializing]
[2022/08/30 14:32:35.483 +00:00] [WARN] [roles/healthz_handler.go:100] ["component is unhealthy"] [state=Initializing]
[2022/08/30 14:32:45.483 +00:00] [WARN] [roles/healthz_handler.go:100] ["component is unhealthy"] [state=Initializing]
[2022/08/30 14:32:46.884 +00:00] [WARN] [roles/healthz_handler.go:100] ["component is unhealthy"] [state=Initializing]
[2022/08/30 14:32:55.484 +00:00] [WARN] [roles/healthz_handler.go:100] ["component is unhealthy"] [state=Initializing]
[2022/08/30 14:32:55.484 +00:00] [WARN] [roles/healthz_handler.go:100] ["component is unhealthy"] [state=Initializing]
[2022/08/30 14:33:05.483 +00:00] [WARN] [roles/healthz_handler.go:100] ["component is unhealthy"] [state=Initializing]
[2022/08/30 14:33:15.484 +00:00] [WARN] [roles/healthz_handler.go:100] ["component is unhealthy"] [state=Initializing]
[2022/08/30 14:33:25.483 +00:00] [WARN] [roles/healthz_handler.go:100] ["component is unhealthy"] [state=Initializing]
[2022/08/30 14:33:25.484 +00:00] [WARN] [roles/healthz_handler.go:100] ["component is unhealthy"] [state=Initializing]
[2022/08/30 14:33:35.484 +00:00] [WARN] [roles/healthz_handler.go:100] ["component is unhealthy"] [state=Initializing]
[2022/08/30 14:33:45.483 +00:00] [WARN] [roles/healthz_handler.go:100] ["component is unhealthy"] [state=Initializing]
[2022/08/30 14:33:55.483 +00:00] [WARN] [roles/healthz_handler.go:100] ["component is unhealthy"] [state=Initializing]
[2022/08/30 14:33:55.483 +00:00] [WARN] [roles/healthz_handler.go:100] ["component is unhealthy"] [state=Initializing]
[2022/08/30 14:33:55.484 +00:00] [WARN] [roles/healthz_handler.go:100] ["component is unhealthy"] [state=Initializing]
[2022/08/30 14:33:55.511 +00:00] [ERROR] [roles/roles.go:524] ["Get signal to exit\n"] [signal=terminated] [stack="github.com/milvus-io/milvus/cmd/roles.(*MilvusRoles).Run\n\t/go/src/github.com/milvus-io/milvus/cmd/roles/roles.go:524\ngithub.com/milvus-io/milvus/cmd/milvus.(*run).execute\n\t/go/src/github.com/milvus-io/milvus/cmd/milvus/run.go:111\ngithub.com/milvus-io/milvus/cmd/milvus.RunMilvus\n\t/go/src/github.com/milvus-io/milvus/cmd/milvus/milvus.go:60\nmain.main\n\t/go/src/github.com/milvus-io/milvus/cmd/main.go:26\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:225"]
[2022/08/30 14:33:55.511 +00:00] [DEBUG] [querycoord/service.go:283] ["QueryCoord stop"] [Address=10.102.8.152:19531]
[2022/08/30 14:33:55.511 +00:00] [INFO] [querycoord/query_coord.go:276] ["close channel unsubscribe loop..."]
[2022/08/30 14:33:55.511 +00:00] [INFO] [querycoord/query_coord.go:281] ["cancel the loop of QueryCoord..."]
[2022/08/30 14:33:55.511 +00:00] [INFO] [querycoord/query_coord.go:285] ["Query Coord stopped successfully..."]
[2022/08/30 14:33:55.511 +00:00] [DEBUG] [querycoord/service.go:295] ["Graceful stop grpc server..."]

```

**Comment by zhuwenxing** [comment]
Now, it is not that stable. Sometimes fail, but sometimes success

success: https://qa-jenkins.milvus.io/blue/organizations/jenkins/chaos-test-for-release/detail/chaos-test-for-release/41/pipeline

failed: https://qa-jenkins.milvus.io/blue/organizations/jenkins/chaos-test-for-release/detail/chaos-test-for-release/24/pipeline

**Comment by zhuwenxing** [comment]
This fix PR https://github.com/milvus-io/milvus/pull/18801 only merged in master branch, but not merged in 2.1.0 branch

**Comment by zhuwenxing** [comment]
not reproduced yet, so close it
